### PR TITLE
Reduce interference with script tag filtering

### PR DESCRIPTION
**Pull request by brandonpayton**
When we encounter a single-script concat group, we aren't actually
concatenating, yet because the script is expected to be concatenated,
we handle printing the script rather than relying on the do_item()
method. This means that the core `script_loader_tag` filter is not
applied to the script tag even though it is still a normal,
single-script tag.

@josephscott [recently discovered](https://github.com/Automattic/page-optimize/pull/44/) that this interfere's with
the TwentyTwenty theme adding async/defer attributes to its scripts
even though they are not being concatenated with anything else.

We address this by manually applying the `script_loader_tag` filter
when we are printing the script tag for a single-script concat group.

### CODE REVIEW

**Comment by brandonpayton** [comment]
I haven't tested this at all yet, but I believe this is a more general solution to #44.

**Comment by josephscott** [comment]
I tried this out and the `async` tag was still omitted.  The PHP error log mentioned:

```
[22-Jun-2020 22:03:11 UTC] PHP Warning:  Illegal offset type in isset or empty in /srv/htdocs/__wp__/wp-includes/class.wp-dependencies.php on line 272
[22-Jun-2020 22:03:11 UTC] PHP Warning:  Illegal offset type in isset or empty in /srv/htdocs/__wp__/wp-includes/class.wp-dependencies.php on line 272
```

**Comment by brandonpayton** [comment]
> I tried this out and the `async` tag was still omitted. 

Thanks for checking it out. It looks like there is at least a bug with the handle that is passed to the filter. The current code under this PR is passing the handles array rather than the first handle in the array.

**Comment by brandonpayton** [comment]
> It looks like there is at least a bug with the handle that is passed to the filter.

OK, with that corrected, it looks like this fix works.

Btw, @josephscott, it's really cool that you noticed this edge case. On my test site, the script isn't standalone and had to be manipulated to "concat" as standalone.

**Comment by josephscott** [comment]
Confirmed, on my test site the `async` attribute is now preserved for `index.js` from the twentytwenty theme.

**Comment by brandonpayton** [comment]
> Confirmed, on my test site the async attribute is now preserved for index.js from the twentytwenty theme.

Thank you! @josephscott, how does the code look to you? Would you be up for reviewing?

**Review by josephscott — APPROVED** [review]
This fixes the problem I was having with index.js missing the async attribute.  Not seeing any other issues so far.

**Comment by josephscott** [comment]
It looks good to me, and I haven't seen any other issues pop up.

May want to slowly try this out on a few other Atomic sites before doing a full roll out.

**Comment by brandonpayton** [comment]
Good idea :)  I'll do a bit more testing

**Comment by brandonpayton** [comment]
Tested with more plugins enabled and specifically with a couple of plugins that use the `script_loader_tag` filter:

* Async JavaScript
* WPO Tweaks & Optimizations

File: concat-js.php [hunk 1]
```diff
@@ -249,10 +249,20 @@ function do_items( $handles = false, $group = false ) {
...
 					if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
-						echo "<script data-handles='" . esc_attr( $handles ) . "' $load_mode type='text/javascript' src='$href'></script>\n";
+						$tag = "<script data-handles='" . esc_attr( $handles ) . "' $load_mode type='text/javascript' src='$href'></script>\n";
 					} else {
-						echo "<script type='text/javascript' $load_mode src='$href'></script>\n";
+						$tag = "<script type='text/javascript' $load_mode src='$href'></script>\n";
 					}
+
+					if ( is_array( $js_array['handles'] ) && count( $js_array['handles'] ) === 1 ) {
+						// Because we have a single script, let's apply the `script_loader_tag` filter as core does in `do_item()`.
```
Comment (aidvu) [inline concat-js.php:258]: Nice! Good job Brandon.
